src/Form/Form.js

```diff
@@ -1,21 +1,37 @@
 const FormAnalyzer = require('./FormAnalyzer')
-const {SUBMIT_BUTTON_SELECTOR, FORM_ELS_SELECTOR} = require('./selectors')
 const {addInlineStyles, removeInlineStyles, setValue, isEventWithinDax, isMobileApp, isApp} = require('../autofill-utils')
-const {getInputSubtype, setInputType, getInputMainType} = require('./input-classifiers')
+const {getInputSubtype, getInputMainType} = require('./matching')
 const {getIconStylesAutofilled, getIconStylesBase} = require('./inputStyles')
 const {ATTR_AUTOFILL} = require('../constants')
 const getInputConfig = require('./inputTypeConfig.js')
 const {getUnifiedExpiryDate, formatCCYear, getCountryName} = require('./formatters')
+const {Matching} = require('./matching')
+const {matchingConfiguration} = require('./matching-configuration')
 
 class Form {
-    constructor (form, input, deviceInterface) {
+    /**
+     * @type {import("./matching").Matching}
+     */
+    matching;
+    /**
+     * @type {HTMLFormElement}
+     */
+    form;
+    /**
+     * @param {HTMLFormElement} form
+     * @param {HTMLInputElement|HTMLSelectElement} input
+     * @param {InterfacePrototypeBase} deviceInterface
+     * @param {Matching} [matching]
+     */
+    constructor (form, input, deviceInterface, matching) {
         this.form = form
-        this.formAnalyzer = new FormAnalyzer(form, input)
+        this.matching = matching || new Matching(matchingConfiguration)
+        this.formAnalyzer = new FormAnalyzer(form, input, matching)
         this.isLogin = this.formAnalyzer.isLogin
         this.isSignup = this.formAnalyzer.isSignup
         this.device = deviceInterface
 
-        /** @type Object<'all' | SupportedMainTypes, Set> */
+        /** @type Record<'all' | SupportedMainTypes, Set> */
```

Comment:
Just to clarify, my understanding is that in this situation `Object` and `Record` are effectively the same, is that right?

Reply:
Not quite - Object inherits all of `Object`'s prototype - whereas `Record` is a strict key-binding.